archive/logic_puzzles/knights_and_knaves.lean

```diff
@@ -0,0 +1,74 @@
+import tactic
+
+/-
```

Comment:
```suggestion
/-!
```
Please add a copyright header to this and the other file. See the guidelines here: https://leanprover-community.github.io/contribute/style.html#header-and-imports and also the next section on module doc strings.

Generally we surround descriptive / expository text with module doc strings `/-! -/`. Ordinary doc strings surrounded by `/- -/` are generally reserved for discussions of implementation details.

See https://github.com/leanprover-community/mathlib/blob/master/archive/sensitivity.lean for example.

Reply:
Thanks for pointing these out Bryan, and for your patience with my first PR. I hope this is OK now after my last edits.